src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs

```diff
@@ -410,55 +410,61 @@ private static ModuleInfo[] GetModuleInfos(int processId, bool firstModuleOnly)
                     if (moduleCount <= moduleHandles.Length) break;
                     moduleHandles = new IntPtr[moduleHandles.Length * 2];
                 }
-                List<ModuleInfo> moduleInfos = new List<ModuleInfo>();
 
-                int ret;
+                List<ModuleInfo> moduleInfos = new List<ModuleInfo>(moduleCount);
+                StringBuilder baseName = new StringBuilder(1024);
+                StringBuilder fileName = new StringBuilder(1024);
+
                 for (int i = 0; i < moduleCount; i++)
                 {
-                    try
+                    while (true)
```

Comment:
It's fine to move the Clears to the top; the only impact there is we'd be clearing when it was already cleared, which should be very cheap.  I didn't understand why we'd need to move and augment the firstModuleOnly check to the top.  Can you elaborate?